topdown/uuid.go

```diff
@@ -31,6 +32,32 @@ func builtinUUIDRFC4122(bctx BuiltinContext, operands []*ast.Term, iter func(*as
 	return iter(result)
 }
 
+func builtinUUIDParse(_ BuiltinContext, operands []*ast.Term, iter func(term *ast.Term) error) error {
+	str, err := builtins.StringOperand(operands[0].Value, 1)
+	if err != nil {
+		return err
+	}
+
+	parsed := uuid.Parse(string(str))
+	if len(parsed) == 0 {
+		return nil
+	}
+
+	result := ast.NewObject()
+	for prop, val := range parsed {
+		switch parsed[prop].(type) {
+		case string:
+			result.Insert(ast.StringTerm(prop), ast.StringTerm(val.(string)))
+		case int:
+			result.Insert(ast.StringTerm(prop), ast.IntNumberTerm(val.(int)))
+		case int64:
+			result.Insert(ast.StringTerm(prop), ast.IntNumberTerm(int(val.(int64))))
```

Comment:
I think the year 2038 is for unix stored as seconds using a signed 32-bit integer. However, I think that unixNano will already overflow with todays date. At time of writing, the unix time is 1693809950, which in ns is `1 693 809 950 000 000 000`, which is `0b1011110000001100111110010110011100010110001011010110000000000`, or 61 bits. So int, assuming it uses 32 bits, will definitely overflow.

Instead, something like this can be done: https://github.com/open-policy-agent/opa/blob/main/topdown/time.go#L89
```go
ast.NumberTerm(int64ToJSONNumber(int64(value)))
```

Reply:
You are completely correct. Didn't look at the sheer size of ns. That is way over 32bits 😮 Yep seems like the `int64toJSONNumber(val int64)` will work. I'll change the insert